File: Data Structures & Algorithms/buy-and-sell-crypto/submission-0.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        max_profit = 0
        min_price = prices[0]

        for i in prices:
            max_profit = max(max_profit, i - min_price)
            min_price = min(min_price, i)
        
        return max_profit

File: Data Structures & Algorithms/valid-binary-search-tree/submission-0.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

public class Solution {
    public bool IsValidBST(TreeNode root) {
        return ValidCheck(root, int.MaxValue, int.MinValue);
    }

    private bool ValidCheck(TreeNode curr, int max, int min) {
        if (curr == null) {
            return true;
        } else if (curr.val >= max || curr.val <= min) {
            return false;
        } else {
            return ValidCheck(curr.left, curr.val, min) &&
                ValidCheck(curr.right, max, curr.val);
        }
    }
}
